cql3/statements/role-management-statements.cc

```diff
@@ -68,11 +68,23 @@ using result_message = cql_transport::messages::result_message;
 using result_message_ptr = ::shared_ptr<result_message>;
 
 static auth::authentication_options extract_authentication_options(const cql3::role_options& options) {
+    static std::string_view allowed_role_options[] = {"read_timeout", "write_timeout"};
+
     auth::authentication_options authen_options;
     authen_options.password = options.password;
 
     if (options.options) {
-        authen_options.options = std::unordered_map<sstring, sstring>(options.options->begin(), options.options->end());
+        authen_options.options.emplace();
+        for (auto& entry : *options.options) {
+            if (std::find(std::begin(allowed_role_options), std::end(allowed_role_options), std::string_view(entry.first)) != std::end(allowed_role_options)) {
```

Comment:
contains()

Reply:
`allowed_role_options` is an array now, due to having just 2 values, so I don't have a `contains` available. Rust traits would be handy here, but in C++ I think it doesn't get prettier than `std::find`.